### PR TITLE
Remove unused method `SearchUtils._get_comparison_for_model_registry`

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Clean model registry searching code

## How is this patch tested?

Unit tests.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/search_utils.py [hunk 1]
```diff
@@ -677,42 +677,6 @@ def _parse_list_from_sql_token(cls, token):
                 error_code=INVALID_PARAMETER_VALUE,
             )
 
-    @classmethod
-    def _get_comparison_for_model_registry(cls, comparison, valid_search_keys):
```
Comment (WeichenXu123) [inline mlflow/utils/search_utils.py:681]: The method is not used after tag based model / version search PR merged.

Comment (harupy) [reply mlflow/utils/search_utils.py:681]: Or you could say title the PR `Remove unused _get_comparison_for_model_registry`.

**Comment by harupy** [comment]
Are there any other unused methods?

**Comment by WeichenXu123** [comment]
I think no.

**Comment by harupy** [comment]
Is `_process_statement_tokens` used?

https://github.com/mlflow/mlflow/blob/80564db826e5a4ce9d1a18eee36d252b03e1ab6c/mlflow/utils/search_utils.py#L723

**Comment by harupy** [comment]
Is `_get_comparison_for_model_registry` used?

https://github.com/mlflow/mlflow/blob/80564db826e5a4ce9d1a18eee36d252b03e1ab6c/mlflow/utils/search_utils.py#L681

**Comment by harupy** [comment]
Is `_is_list_component_token` used?

https://github.com/mlflow/mlflow/blob/80564db826e5a4ce9d1a18eee36d252b03e1ab6c/mlflow/utils/search_utils.py#L717

**Review by dbczumar — APPROVED** [review]
LGTM!

**Pull request merged**